homeassistant/components/roomba/__init__.py

```diff
@@ -146,10 +147,16 @@ async def async_connect_or_timeout(hass, roomba):
 
 async def async_disconnect_or_timeout(hass, roomba):
     """Disconnect to vacuum."""
-    _LOGGER.debug("Disconnect vacuum")
-    with async_timeout.timeout(3):
-        await hass.async_add_job(roomba.disconnect)
-    return True
+    try:
+        with async_timeout.timeout(3):
+            _LOGGER.debug("Disconnect vacuum")
+            await hass.async_add_executor_job(roomba.disconnect)
+            while roomba.roomba_connected:
+                await asyncio.sleep(1)
+                break
```

Comment:
This will just sleep once for 1 second. We don't need the while loop if that is what we want.

Reply:
What should happen here?

Reply:
In fact, the problem is this: the roombapy API makes your hand too early. I have to wait for the disconnection, moreover if we reconnect too quickly, the same thing happens, the connection cannot be initiated.

So I'm waiting for a disconnection and I wanted to pause for a second after the disconnection to be able to reconnect. Because the async_reload function is too fast